File: app/src/main/java/com/template/mvvmtemplate/error/ErrorManagement.java
package com.template.mvvmtemplate.error;

import android.content.Context;
import android.util.Log;
import android.widget.Toast;

import com.template.mvvmtemplate.BuildConfig;

/**
 * Created by diegonovati on 18/04/2018.
 */

/**
 * Class that manages the errors raised by the App. On the debug version the errors are displayed to
 * the user, on the production version the error are sent to Crashlytics.
 */
public final class ErrorManagement {

    private static final String TAG = "ErrorManagement";

    public static void logError(Context context, Throwable ex) {
        logError(context, ex, true);
    }

    public static void logError(Context context, Throwable ex, boolean showMessage) {
        Log.e(TAG, "logError: ", ex);

        try {
            if (showMessage && BuildConfig.displayErrorMessage) {
                Toast.makeText(context.getApplicationContext(), ex.getMessage(), Toast.LENGTH_SHORT).show();
                return;
            }
            // Add here the code to send the error to Crashlytics
        } catch (Exception e) {
            Log.e(TAG, "******** logError: ", e);
        }
    }
}

File: app/src/main/java/com/template/mvvmtemplate/error/ErrorManagement.kt
package com.template.mvvmtemplate.error

import android.content.Context
import android.util.Log
import android.widget.Toast

import com.template.mvvmtemplate.BuildConfig

/**
 * Class that manages the errors raised by the App. On the debug version the errors are displayed to
 * the user, on the production version the error are sent to Crashlytics.
 */
object ErrorManagement {

    private const val TAG = "ErrorManagement"

    @JvmOverloads
    fun logError(ex: Throwable, context: Context? = null, showMessage: Boolean = true) {
        Log.e(TAG, "logError: ", ex)

        try {
            if (showMessage && BuildConfig.displayErrorMessage && context != null) {
                Toast.makeText(context.applicationContext, ex.message, Toast.LENGTH_SHORT).show()
            } else {
                // Add here the code to send the error to Crashlytics
            }
        } catch (e: Exception) {
            Log.e(TAG, "******** logError: ", e)
        }
    }
}
